src/v/storage/disk_log_impl.cc

```diff
@@ -1009,7 +1009,9 @@ gc_config disk_log_impl::maybe_override_retention_config(gc_config cfg) const {
      * don't override with local retention settings--let partition data expand
      * up to standard retention settings.
      */
-    if (!config::shard_local_cfg().retention_local_strict()) {
+    if (
+      !config::shard_local_cfg().retention_local_strict()
+      && !config::shard_local_cfg().retention_local_strict_override()) {
```

Comment:
i think the test is failing as now we must set both properties if we do not want to use strict local retention, is that expected ?

Reply:
My mistake, I believe if should be one or the other.